test-data/unit/lib-stub/functools.pyi

```diff
@@ -0,0 +1,35 @@
+from typing import Generic, TypeVar, Callable, Any, Mapping
```

Comment:
Some tests were using the typeshed stub for `functools`, and it requires `typing._SpecialForm`. Instead of adding a bunch of `typing-medium.pyi` fixtures, I decided to create a simplified stub to use in tests. This also speeds up the test cases a bit.